heroic-core/src/main/java/com/spotify/heroic/metric/LocalMetricManager.java

```diff
@@ -340,12 +375,22 @@ public QueryTrace buildTrace() {
 
             // Transform that takes the result from ES metadata lookup to fetch from backend
             final LazyTransform<FindSeries, FullQuery> transform =
-                new Transform(request, failOnLimits, seriesLimit, groupLimit, quotaWatcher,
-                    dataInMemoryReporter);
+                new Transform(request,
+                    failOnLimits,
+                    seriesLimit,
+                    groupLimit,
+                    quotaWatcher,
+                    dataInMemoryReporter,
+                    span);
+
+            final Span metadataSpan = tracer.spanBuilderWithExplicitParent(
+                "localMetricsManager.findSeries", span).startSpan();
 
             return metadata
-                .findSeries(
-                    new FindSeries.Request(request.filter(), request.range(), seriesLimit))
+                .findSeries(new FindSeries.Request(request.filter(), request.range(), seriesLimit))
+                .onResolved(t -> metadataSpan.putAttribute(
+                    "seriesCount", longAttributeValue(t.getSeries().size())))
+                .onDone(new EndSpanFutureReporter(metadataSpan))
```

Comment:
If I'm reading the async source correctly, runnables on the future get executed in reverse order that they're set. Which would mean the transform is happening before metadataSpan ends, causing the incorrect timing you screenshotted. 

Check my math on that though: https://github.com/udoprog/tiny-async-java/blob/1.11.0/tiny-async-core/src/main/java/eu/toolchain/async/concurrent/ConcurrentResolvableFuture.java#L491

Reply:
I moved the closing of the spans around, moving it to the bottom where `.onDone(reporter.reportQueryMetrics());` is produces...

![Screen Shot 2019-10-16 at 10 26 51 AM](https://user-images.githubusercontent.com/677960/66928507-7dee9f00-efff-11e9-86a9-53385ff583f9.png)

which makes the `findSeries` the entire length of the `fetch`


Adding it back to where it was actually shows some spans closing at the current timing

![Screen Shot 2019-10-16 at 10 27 37 AM](https://user-images.githubusercontent.com/677960/66928612-abd3e380-efff-11e9-99b3-046f64eaad1a.png)


vs others...

![Screen Shot 2019-10-16 at 10 27 44 AM](https://user-images.githubusercontent.com/677960/66928620-b0000100-efff-11e9-81ee-9b3f1ce456c0.png)

very strange...